framework/include/materials/Material.h

```diff
@@ -244,10 +256,8 @@ class Material : public MooseObject,
   /// If False MOOSE does not compute this property
   const bool _compute;
 
-  /// False by default.  If true, MOOSE will only call
-  /// computeQpProperties() for the 0th qp and then copy
-  /// that value around to all the qps.
-  const bool _constant_on_elem;
+  /// Options of the constantness level of the material
+  const int _constant_option;
```

Comment:
Oh yeah, you are using a raw `int` here. Really, you should change this to an enum type.